staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/controller_ref.go

```diff
@@ -0,0 +1,41 @@
+package v1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// IsOwnedBy checks if the  object has a controllerRef set to the given owner
+func IsOwnedBy(obj Object, owner Object) bool {
+	ref := GetControllerOf(obj)
+	if ref == nil {
+		return false
+	}
+	return ref.UID == owner.GetUID()
+}
+
+// GetControllerOf returns the controllerRef if controllee has a controller,
+// otherwise returns nil.
+func GetControllerOf(controllee Object) *OwnerReference {
+	ownerRefs := controllee.GetOwnerReferences()
+	for i := range ownerRefs {
+		owner := &ownerRefs[i]
+		if owner.Controller != nil && *owner.Controller == true {
+			return owner
+		}
+	}
+	return nil
+}
```

Comment:
Yes, I understand what you mean :)
The difference is that my version returns a pointer to a copy (as you correctly pointed out) and the original version returns a pointer to the value stored in the slice. However in this case it does not matter in practice - there are only four implementations of `GetOwnerReferences()` - all of them return copies of the internal owner references slice. So the code that called `GetControllerOf()` cannot mutate the in-object owner references in any case because it will gat a pointer to a copy.